sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileAsyncClient.java

```diff
@@ -0,0 +1,437 @@
+     * Docs</a></p>
+     *
+     * @param position The length of the file after all data has been written.
+     *
+     * @return A reactive response containing the information of the created resource.
+     */
+    public Mono<PathInfo> flush(long position) {
+        try {
+            return flushWithResponse(position, false, false, null, null).flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Flushes (writes) data previously appended to the file through a call to append.
+     * The previously uploaded data must be contiguous.
+     *
+     * <p><strong>Code Samples>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileAsyncClient.flushWithResponse#long-boolean-boolean-PathHttpHeaders-DataLakeRequestConditions}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update">Azure
+     * Docs</a></p>
+     *
+     * @param position The length of the file after all data has been written.
+     * @param retainUncommittedData Whether or not uncommitted data is to be retained after the operation.
+     * @param close Whether or not a file changed event raised indicates completion (true) or modification (false).
+     * @param httpHeaders {@link PathHttpHeaders httpHeaders}
+     * @param accessConditions {@link DataLakeRequestConditions accessConditions}
+     *
+     * @return A reactive response containing the information of the created resource.
+     */
+    public Mono<Response<PathInfo>> flushWithResponse(long position, boolean retainUncommittedData, boolean close,
+        PathHttpHeaders httpHeaders, DataLakeRequestConditions accessConditions) {
+        try {
+            return withContext(context -> flushWithResponse(position, retainUncommittedData, close, httpHeaders,
+                accessConditions, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<PathInfo>> flushWithResponse(long position, boolean retainUncommittedData, boolean close,
+        PathHttpHeaders httpHeaders, DataLakeRequestConditions accessConditions, Context context) {
+
+        httpHeaders = httpHeaders == null ? new PathHttpHeaders() : httpHeaders;
+        accessConditions = accessConditions == null ? new DataLakeRequestConditions() : accessConditions;
+
+        LeaseAccessConditions lac = new LeaseAccessConditions().setLeaseId(accessConditions.getLeaseId());
+        ModifiedAccessConditions mac = new ModifiedAccessConditions()
+            .setIfMatch(accessConditions.getIfMatch())
+            .setIfNoneMatch(accessConditions.getIfNoneMatch())
+            .setIfModifiedSince(accessConditions.getIfModifiedSince())
+            .setIfUnmodifiedSince(accessConditions.getIfUnmodifiedSince());
```

Comment:
We should probably add an item to update the swagger/readme so we can just get rid of these types.

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/6105